lib/src/logic/questionnaire_controller.dart

```diff
@@ -630,8 +631,7 @@ class QuestionnaireController {
             ? null
             : [
                 QuestionnaireResponseAnswer(
-                    valueBoolean:
-                        FhirBoolean(itemBundle.controller.rawValue as bool))
+                    valueX: FhirBoolean(itemBundle.controller.rawValue as bool))
               ];
         break;
       case QuestionnaireItemType.choice:
```

Comment:
This case is using a `QuestionnaireItemType.choice` instead of `QuestionnaireItemTypeEnum.choice`, again no semantic issues with the types, the `itemType` being switched is `final itemType = itemBundle.item.type.valueEnum;` so all cases must be of type enum. This is silently failing.